GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/GameManager.cs

```diff
@@ -0,0 +1,140 @@
+    public GameObject garagePageCanvas;
+    public Text coinsCount;
+    public GameObject storeItemCarSedanGameObj;
+    public GameObject storeItemCarTruckGameObj;
+    public GameObject storeItemCarJeepGameObj;
+    public GameObject storeItemCarKartGameObj;
+    public GameObject garageItemCarSedanGameObj;
+    public GameObject garageItemCarTruckGameObj;
+    public GameObject garageItemCarJeepGameObj;
+    public GameObject garageItemCarKartGameObj;
+    public GameObject playCarSedanGameObj;
+    public GameObject playCarJeepGameObj;
+    public GameObject playCarTruckGameObj;
+    public GameObject playCarKartGameObj;
+
+    private const string Filename = "data.json";
+    private string _dataPath;
+    private GameData _gameData;
+    private List<GameObject> _canvasPagesList;
+
+    // init the game
+    public void Awake()
+    {
+        // user login
+        _dataPath = Application.persistentDataPath + "/" + Filename;
+        Debug.Log(_dataPath);
+        InitCarList();
+        LoadGameData();
+        SetCoins();
+        _canvasPagesList = new List<GameObject>() {playPageCanvas, storePageCanvas, garagePageCanvas};
+    }
+
+    // set the coins count at the play page
+    public void SetCoins()
+    {
+        coinsCount.text = _gameData.coinsOwned.ToString();
+    }
+
+    // switch pages when enter the store.
+    public void OnEnterStoreButtonClicked()
+    {
+        SetCanvas(storePageCanvas);
+    }
+
+    // switch pages when enter the play.
+    public void OnEnterPlayPageButtonClicked()
+    {
+        SetCanvas(playPageCanvas);
+    }
+
+    // switch pages when enter the garage.
+    public void OnEnterGaragePageButtonClicked()
+    {
+        SetCanvas(garagePageCanvas);
+    }
+
+    private void SetCanvas(GameObject targetCanvasPage)
+    {
+        // set all canvas pages to be inactive
+        foreach (var canvasPage in _canvasPagesList)
+        {
+            canvasPage.SetActive(false);
+        }
+
+        // set the target canvas page to be active
+        targetCanvasPage.SetActive(true);
+    }
+
+    // check if the player is in play mode (page)
+    public bool IsInPlayPage()
+    {
+        return playPageCanvas.activeInHierarchy;
+    }
+
+    // save game data
+    public void SaveGameData()
+    {
+        System.IO.File.WriteAllText(_dataPath, JsonUtility.ToJson(_gameData, true));
+    }
+
+    // load game data
+    private void LoadGameData()
+    {
+        try
+        {
+            // check if the data file exits
+            if (System.IO.File.Exists(_dataPath))
+            {
+                var contents = System.IO.File.ReadAllText(_dataPath);
+                _gameData = JsonUtility.FromJson<GameData>(contents);
+            }
+            else // if data file doesn't exist, create a default one
+            {
+                Debug.Log("Unable to read the save data, file does not exist");
+                _gameData = new GameData();
+                SaveGameData();
+            }
```

Comment:
Let game data manage this by itself. You can instantiate a gameData object, then, in its constructor, it will try to load data by doing this IO operation.
Every time you make update to game data, it will save data by itself.
As a future work: 
Since it involves disk operation, you can add a check isReady to gameData. For now, let it block the main thread but later on you can set up some async operation.

Reply:
I was trying to do this. However, I am not able to use _gameData = JsonUtility.FromJson<GameData>(contents) in that way. I cannot load the object itself from a json file from inside the object.